### PR TITLE
Excel export

### PR DESCRIPTION
**Pull request by manuelblum**
Exporting table data available with this Mr. Used package [exceljs](https://github.com/exceljs/exceljs) for creating excel sheet.

### Main Features :
- ExportExcelApi (new api for table) 
- export current displayed Page - useExportDisplayedTableData() - this hook will export the currently displayed data from the attached Table
- export more data from table (especially paged) - useExportTableQuery() - this hook will query moreData from the attached Table and the query parameter with an apollo client)
- ExcelExportButton which takes an exportApi and triggers an export
Screenshot 2019-11-26 at 12 49 54

### Examples

**Export Current Page:**
[TableExcelExport.tsx](https://github.com/vivid-planet/react-admin/blob/excel-export/packages/react-admin-stories/src/react-admin-core/TableExcelExport.tsx)

**Exported xlsx fill look like:**
Screenshot 2019-11-26 at 12 45 29

**Export All Pages:**
 [TablePagingExcelExport.tsx](https://github.com/vivid-planet/react-admin/blob/excel-export/packages/react-admin-stories/src/react-admin-core/TablePagingExcelExport.tsx)

**Exported xlsx fill look like:**
Screenshot 2019-11-26 at 13 26 22

### CODE REVIEW

File: packages/react-admin-core/src/table/TableQueryContext.tsx [hunk 1]
```diff
@@ -1,9 +1,12 @@
 import { DocumentNode } from "graphql";
 import * as React from "react";
+import { ITableData } from "./useTableQuery";
 
 export interface ITableQueryApi {
     getVariables: () => object;
+    getInnerOptions: () => object;
     getQuery: () => DocumentNode;
+    getResolveTableData: (data: object) => ITableData<any>;
```
Comment (nsams) [inline packages/react-admin-core/src/table/TableQueryContext.tsx]: warum get? (weil das "resolved" ja praktisch gleich)

File: packages/react-admin-core/src/table/excelexport/useExportTableQuery.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import { Table } from "../Table";
+import { ITableQueryApi } from "../TableQueryContext";
+import { createExcelExportDownload } from "./createExcelExportDownload";
+import { IExportApi } from "./IExportApi";
+
+export function useExportTableQuery<IVariables>(
+    api: ITableQueryApi,
+    variables: IVariables,
+    fileName?: string,
+    worksheetName?: string,
```
Comment (nsams) [inline packages/react-admin-core/src/table/excelexport/useExportTableQuery.tsx]: da wär ich ja dafür teile (oder alles) in options objekt zu packen

Comment (manuelblum) [reply packages/react-admin-core/src/table/excelexport/useExportTableQuery.tsx]: das hab ich auf resolveTableData geändert

File: packages/react-admin-core/src/table/TableQueryContext.tsx (hunk 1 above)
Comment (manuelblum) [reply packages/react-admin-core/src/table/TableQueryContext.tsx]: resolveTableData

**Pull request merged**